src/librustc_mir/borrow_check/conflict_errors.rs

```diff
@@ -614,8 +614,9 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                     projection,
                 } = first_borrowed_place;
 
-                for (i, elem) in projection.iter().enumerate().rev() {
-                    let proj_base = &projection[..i];
+                let mut cursor = &**projection;
```

Comment:
Can we refactor this idiom into a method on `projection` to reduce the repetition?